reactive/docker.py

```diff
@@ -330,11 +331,21 @@ def remove_nrpe_config(nagios=None):
         nrpe_setup.remove_check(shortname=service)
 
 
+class ConfigError(Exception):
+    pass
+
+
+def validate_config():
+    '''Check that config is valid.'''
+    if len(config('no_proxy')) + len("Environment=\"NO_PROXY=\"\"") > 2048:
+        raise ConfigError('no_proxy longer than 2023 characters.')
```

Comment:
What about doing a status_set() here too so the operator can easily see what's wrong? Also, nitpick but the 2023 should be calculated, not hard-coded.

Reply:
The status_set was my initial goal as well. Unfortunately this would be a much more involving patch since the status message is overwritten by the ready status on the kubernetes-worker top layer: https://github.com/kubernetes/kubernetes/blob/master/cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py#L272 (and probably somewhere in the top layer of canal)

Reply:
Okay, let's take the quick win and move on - thanks for the explanation.